docs/latest/testing/index.md

```diff
@@ -181,27 +188,63 @@ For testing client-side island behavior (clicks, state changes, etc.), you need
 a full build and browser environment. You can use the approach similar to
 Fresh's own tests:
 
-```tsx island-client.test.tsx
+```tsx tests/island-client.test.tsx
 import { expect } from "@std/expect";
-import { createBuilder } from "vite";
+import { buildFreshApp, startTestServer } from "./test-utils.ts";
+
+const app = await buildFreshApp();
+
+Deno.test("Counter island renders correctly", async () => {
+  const { server, address } = startTestServer(app);
+
+  try {
+    // Basic smoke test: verify the island HTML is served
+    const response = await fetch(`${address}/`);
+    const html = await response.text();
+
+    expect(html).toContain('class="flex gap-8 py-6"');
+    expect(html).toContain("3");
+
+    // For full browser interactivity testing, you would need:
+    // - Browser automation tools (Puppeteer, Playwright)
+    // - withBrowser utility from Fresh's test suite
+  } finally {
+    await server.shutdown();
+  }
+});
+```
+
+```tsx tests/test-utils.ts
+import { createBuilder, type InlineConfig } from "vite";
 import * as path from "@std/path";
 
-// Create a production build
-const builder = await createBuilder({
+// Default Fresh build configuration
+export const FRESH_BUILD_CONFIG: InlineConfig = {
   logLevel: "error",
   root: "./",
   build: { emptyOutDir: true },
   environments: {
     ssr: { build: { outDir: path.join("_fresh", "server") } },
     client: { build: { outDir: path.join("_fresh", "client") } },
   },
-});
-await builder.buildApp();
+};
+
+// Helper function to create and build the Fresh app
+export async function buildFreshApp(config: InlineConfig = FRESH_BUILD_CONFIG) {
+  const builder = await createBuilder(config);
+  await builder.buildApp();
+  return await import("../_fresh/server.js");
+}
 
-const app = await import("./_fresh/server.js");
+// Type for Fresh app module
+export interface FreshApp {
+  default: {
+    fetch: (req: Request) => Response | Promise<Response>;
+  };
+}
```

Comment:
Let's inline this into `buildFreshApp`. I think the return type is always a `Promise` too.